apps/main-landing/src/components/products-section/components/prediction-markets-section/prediction-markets-section-actions.tsx

```diff
@@ -0,0 +1,108 @@
+import { Button } from '@idriss-xyz/ui/button';
+import { classes } from '@idriss-xyz/ui/utils';
+import { useMutation } from '@tanstack/react-query';
+import { Controller, useForm } from 'react-hook-form';
+
+type WaitlistInput = {
+  email: string;
+};
+
+export const PredictionMarketsSectionActions = () => {
+  const {
+    control,
+    handleSubmit,
+    reset,
+    formState: { errors },
+  } = useForm<WaitlistInput>({
+    defaultValues: {
+      email: '',
+    },
+  });
+
+  const mutation = useMutation({
+    mutationFn: async (data: WaitlistInput) => {
+      const response = await fetch('https://api.idriss.xyz/subscribe', {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify(data),
+      });
+
+      if (!response.ok) {
+        throw new Error('Failed to join waitlist');
+      }
+
+      return response.json() as Promise<Record<string, string>>;
```

Comment:
can you add TODO comment to validate response schema?

Reply:
actually I don't see that you're using this response data so you can remove this type casting just do `await response.json()` without returning